automatic/mkdocs/tools/ChocolateyInstall.ps1

```diff
@@ -1,10 +1,34 @@
 ﻿Update-SessionEnvironment
-
+$allowed_python_versions = @('3.9', '3.8', '3.7', '3.6', '3.5') # sync with nuspec
 $version = '1.1.2'
 
 $proxy = Get-EffectiveProxy
 if ($proxy) {
   Write-Host "Setting CLI proxy: $proxy"
   $env:http_proxy = $env:https_proxy = $proxy
 }
-python -m pip install mkdocs==$version
+
+function FindPython {
+  param(
+    $allowed_python_versions
+  )
+  # see https://www.python.org/dev/peps/pep-0514/#structure
+  # we are querying machine regsitry only because chocolatey
+  # installs python in admin mode only.
+  $avaiable_installation = Get-ChildItem -Path Registry::HKEY_LOCAL_MACHINE\Software\Python\PythonCore | Select-Object Name
+  foreach ($install in $avaiable_installation) {
+    $name_install = $install.Name
+    $install_version = ($name_install -split '\\')[-1]
+    if ($allowed_python_versions.Contains($install_version)) {
+      Write-Host "Found Python Version from Registry $install_version" -ForegroundColor Yellow
+      $python_executable = Get-ItemProperty -Path "Registry::$name_install\InstallPath" | Select-Object ExecutablePath
+      Write-Host "Found Install Path from Registry $($python_executable.ExecutablePath)" -ForegroundColor Yellow
+      return $python_executable.ExecutablePath
+    }
+  }
```

Comment:
This code will return the first Python executable it finds in the `allowed_python_versions`. If there are multiple versions of Python installed it may not pick the latest?

Reply:
Yeah, it doesn't pick the latest. I am thinking about reversing the array so that the latest version is picked.
```ps1
if ($avaiable_installation.length -gt 1){
    [array]::Reverse($avaiable_installation)
}
```
is this fine?